app/views/recipe_ingredients/new.html.erb

```diff
@@ -0,0 +1,10 @@
+<h1>Add Ingredient to <%= @recipe.name %></h1>
+<%= form_with url: "/recipes/#{@recipe.id}", method: :post, local: true do |f| %>
+  <%= f.label :name, "Ingredient Name" %>
+  <%= f.text_field :name %>
+
+  <%= f.label :cost, "Ingredient Cost" %>
+  <%= f.number_field :cost %>
+
+  <%= f.submit %>
+<% end %>
```

Comment:
Taking a look at this form, I'm not too sure what it's supposed to be doing. Here's the user story:

```plaintext
Extension 2
Add an Ingredient to a Recipe

As a visitor
When I visit '/recipes/:id'
Then I see a form to add an ingredient to this recipe.
When I fill in a field with an existing ingredient's ID,
And I click submit,
Then I am redirrected to the recipe's show page,
And I see the new ingredient listed for this recipe.
```

The form is supposed to take the existing ID of an ingredient but it instead takes a name and a cost and sends those values to `"/recipes/:id"`. If we follow that trail through the routes, ` post "/recipes/:id", to: "recipe_ingredients#create"`, we end up in the `create` action of the `RecipeIngredients` controller where we then create a new ingredient:

```ruby
def create
  recipe = Recipe.find(params[:id])
  recipe.ingredients.create(ingredient_params)

  redirect_to "/recipes/#{recipe.id}"
end
```

The verb, controller, and action are correct in the routes.rb file. I left comments on the path in the routes.rb file. And comments in the controller.